### PR TITLE
DOCSP-19604: mongoclient srv options

### PR DESCRIPTION
**Pull request by ccho-mongodb**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-19604

### Snooty build log:
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?jobId=618d4119c13055282d26ad2b

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-19604-srv-options/fundamentals/connection/#connection-options

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging and workerpool job links in the PR description updated?

### If your page documents a concept, does it meet the following criteria?

- [ ] Target the [Jasmin persona](https://drive.google.com/file/d/14FbBOLCVxwSP6M9BK4Nz1Ir9tzxT8_02/view)
- [ ] Target the [Lucas persona](https://drive.google.com/file/d/1J2vqJxo7ldv7OP_obA9Q-avf0o_ju4Lk/view)

### CODE REVIEW

**Review by terakilobyte — APPROVED** [review]
copy lgtm

**Review by nbbeeken — APPROVED** [review]

File: source/fundamentals/connection.txt [hunk 1]
```diff
@@ -262,6 +268,22 @@ URI to specify the behavior of the client.
        MongoDB manual entry on :manual:`Versioned API </reference/versioned-api/>`
        for more information.
 
+   * - **srvMaxHosts**
```
Comment (dariakp) [inline source/fundamentals/connection.txt:271]: let's clarify that `0` means "unlimited"

**Review by dariakp — CHANGES_REQUESTED** [review]

File: source/fundamentals/connection.txt [hunk 2]
```diff
@@ -262,6 +268,23 @@ URI to specify the behavior of the client.
...
+     - integer
+     - ``0``
+     - | Sets the maximum number of hosts the driver can connect to when using
+       | the DNS seedlist (SRV) connection protocol, identified by the
+       | ``mongodb+srv`` connection string prefix. When set to ``0``, the
+       | driver does not limit the number of hosts.
+       |
+       | Throws an exception if the value is not equal to ``0`` and you are
+         not using the SRV connection protocol
```
Comment (dariakp) [inline source/fundamentals/connection.txt:280]: Just a tiny nit for consistency:
```suggestion
         not using the SRV connection protocol.
```

**Review by dariakp — CHANGES_REQUESTED** [review]

**Review by dariakp — APPROVED** [review]

**Comment by dariakp** [comment]
We merged the implementation and planning to release tomorrow, so this PR should be all clear to go in, thanks! @terakilobyte

**Pull request merged**